engine/consensus/compliance/engine.go

```diff
@@ -105,6 +123,16 @@ func NewEngine(
 	handler := engine.NewMessageHandler(
 		log.With().Str("compliance", "engine").Logger(),
 		engine.NewNotifier(),
+		engine.Pattern{
+			Match: func(msg *engine.Message) bool {
+				_, ok := msg.Payload.(*messages.BlockResponse)
+				if ok {
+					core.metrics.MessageReceived(metrics.EngineCompliance, metrics.MessageBlockResponse)
+				}
+				return ok
+			},
+			Store: pendingRangeResponses,
+		},
```

Comment:
> Updated compliance engine to accept block ranges, this code is on master but I have made a mistake when merging so now it's fixed

I might be missing some context on this change, I'm unsure why we've added this 🤔. Have a few questions:
* Do we ever send range requests to the compliance engine (as opposed to the sync engine)? I can't see this anywhere. In general I'm not sure why we handle this message type in this engine now. 
* Should we make the same change to the collection node compliance engine? It doesn't have a queue for this message type currently
https://github.com/onflow/flow-go/blob/201deb6a6dd923e2d1bc4396ddc82081c5a70833/engine/collection/compliance/engine.go#L85-L104

Reply:
Oh man, [PR 2857](https://github.com/onflow/flow-go/pull/2857)  added so many changes ... I wish they had talked to me before adding all the channels. They could have just listened to the `OnBlockIncorporated` event from HotStuff instead of adding an entirely new mechanism 😞